portfolio/src/main/webapp/script.js

```diff
@@ -12,12 +12,19 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+function getMessages() {
+  fetch('/data').then(response => response.json()).then((messages) => {
+    const messagesElement = document.getElementById('messages-container');
+    console.log(messages);
+    messagesElement.innerHTML = messages[0];
```

Comment:
Can you figure out how to deal with *all* the messages? Also, what would happen if the messages array returned by the servlet were empty? Trying to think ahead for dealing with a dynamic list of messages...

Reply:
I could not figure out how to do this without innerHTML. Any tips? I looped through messages and added it to the element using innerHTML.